lib/Doctrine/Common/DataFixtures/Purger/ORMPurger.php

```diff
@@ -280,4 +282,22 @@ private function getDeleteFromTableSQL(string $tableName, AbstractPlatform $plat
 
         return 'DELETE FROM ' . $tableIdentifier->getQuotedName($platform);
     }
+
+    /**
+     * @param AbstractPlatform $platform
+     * @param Connection       $connection
+     * @param string           $tbl
+     *
+     * @return string
+     */
+    private function getTruncateTableSQL(AbstractPlatform $platform, Connection $connection, $tbl)
+    {
+        $sql = $platform->getTruncateTableSQL($tbl, true);
+
+        if ($connection->getDriver() instanceof AbstractMySQLDriver) {
```

Comment:
A comment explaining in detail why this is needed would be nice.

Reply:
You mean a comment in the code or just here?

I have a test which relies on the auto-incremented ids of a small number of fixtures. I can't purge with delete because that does not reset the auto-increment to 1. I can't purge with truncate, which would reset the auto-increment, because that does not work on MySQL once i have a single relation mapped. In both cases i can only drop & create the database before loading the fixtures again.

Reply:
A code comment could be great.

> because that does not work on MySQL once i have a single relation mapped

Why does it not work? That's what the comment should explain, possibly with references to the MySQL docs.